### ISSUE TITLE
SERVICE ALERT - sysprep failures affecting Windows 2016 / 2019 and Windows 10

### ISSUE THREAD

**Issue by danielsollondon** [danielsollondon/azvmimagebuilder#83]
We have noticed intermittent sysprep failures affecting Windows 2016 / 2019 and Windows 10, we understand the issue, and working to provide mitigation and resolve it, we will update this thread today. Apologies for the inconvenience.

**Comment by danielsollondon** [comment]
Symptoms:
The image build runs, and then it fails to run sys prep, in the **customization.log** you see these errors:

```text
[xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx] PACKER OUT     azure-arm: IMAGE_STATE_UNDEPLOYABLE
[xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx] PACKER OUT     azure-arm: IMAGE_STATE_UNDEPLOYABLE
[xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx] PACKER OUT     azure-arm: IMAGE_STATE_UNDEPLOYABLE
[xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx] PACKER OUT     azure-arm: IMAGE_STATE_UNDEPLOYABLE
[xxxxxx-xxxx-xxxx-xxxx-xxxxxxxxxxxx] PACKER OUT     azure-arm: IMAGE_STATE_UNDEPLOYABLE
```

If you can get the panther.log you see:
```text
2020-10-xx 00:00:00, Info       [0x0f008b] SYSPRP RunRegistryDlls:Found entrypoint in registry at SOFTWARE\Microsoft\Windows\CurrentVersion\Setup\SysPrepExternal\Generalize\cac93847-1974-41b7-85e4-617943cf22bb; will try to launch 'C:\windows\system32\VMAgentDisabler.dll,SysprepProviderEntryPoint'
2020-10-22 00:00:00, Error      [0x0f0085] SYSPRP LaunchDll:Could not load DLL C:\windows\system32\VMAgentDisabler.dll[gle=0x0000007e]
```
We are testing a workaround, the thread will be updated on Sunday.

**Comment by danielsollondon** [comment]
Hi All, a quick update, this issue is being caused by an underlying component, we have been trying to remove it out of the build. Whilst we can now resume successful builds, we are seeing regressions with Win10 images that are created, where it is taking 10mins+ to login for the first time. Therefore we don't want to post the workaround until we understand the cause of the logon time, this will resume tomorrow. In the meantime, an update is being rolled out to the underlying component but will not complete until the end of the week. For the short term, we can only recommend retrying the builds, and we will try to get the workaround completed. Apologies for the inconvenience, we are working on it.

**Comment by danielsollondon** [comment]
Hi All, the underlying component update **estimated** is to complete tomorrow morning Pacific time, we will confirm tomorrow. In parallel we did further test the some mitigations and could get Win 10 built images and fast login using the /mode:vm, however we did not publish these, as we have not completed final review and spent time testing the underlying component. Next update tomorrow.

**Comment by danielsollondon** [comment]
Hi All, the rollout for the underlying component has completed across all impacted regions, we are still testing AIB, but initial tests are showing successful builds, so please restart you're builds.

I will comeback once we have concluded our testing, later today.

**Comment by danielsollondon** [comment]
Hi, we have completed our testing on Win10, Svr2016, Svr 2019, we have seen successful builds and VM creates, I will leave the issue open for a couple of days, please reply here if you still see issues, with the region you are using.

Again, apologies for the inconvenience, we are working on a plan to avoid this type of issue in the future.

Note, with the Win10 Image, we are seeing slow logon without the use of sysprep /mode:vm, we don't believe this was caused by the underlying component, and will escalate to the Windows image team. This will be tracked in another issue, see image details below.

ImagePublisher=MicrosoftWindowsDesktop
ImageOffer=Windows-10
ImageSku=20h1-pro
ImageVersion=19041.572.2010091946

**Comment by danielsollondon** [comment]
closing, no further issues seen.
